io/disk/arcconf/arcconf_migration.py

```diff
@@ -0,0 +1,250 @@
+           len(self.initial_raid) > 1 or self.migration_sleep is '':
+            self.skip(" please ensure yaml parameters are not empty or \
+                       the total device should be more than 1")
+        elif self.comp(self.pci_id, pci_id_formatted) == 1:
+            self.skip(" Test skipped!!, PMC controller not available")
+
+        detected_distro = distro.detect()
+        if not smm.check_installed("arcconf"):
+            if detected_distro.name == "Ubuntu":
+                http_repo = "%s%s.deb" % (self.http_path, self.tool_name)
+                self.repo = self.fetch_asset(http_repo, expire='10d')
+                cmd = "dpkg -i %s" % self.repo
+            else:
+                http_repo = "%s%s.rpm" % (self.http_path, self.tool_name)
+                self.repo = self.fetch_asset(http_repo, expire='10d')
+                cmd = "rpm -ivh %s" % self.repo
+            if process.system(cmd, ignore_status=True, shell=True) == 0:
+                self.skip("Unable to install arcconf")
+
+        self.os_drive = ""
+        cmd = "lsscsi  | grep LogicalDrv | awk \'{print $7}\'"
+        self.os_drive = self.cmdop_list(cmd)
+
+        if self.diskoverwrite == 'Y' and self.os_drive != "":
+            # Delete a logical drive only if it is not OS drive
+            cmd = "df -h /boot | grep %s" % self.os_drive
+            if process.system(cmd, timeout=300, ignore_status=True,
+                              shell=True) == 0:
+                self.skip("Test Skipped!! OS disk requested for removal")
+
+            self.log.info("Deleting the default logical drive %s" %
+                          (self.logicaldrive))
+            cmd = "echo y | arcconf delete %s logicaldrive %s" % \
+                (self.crtl_no, self.logicaldrive)
+            self.check_pass(cmd, "Failed to delete Logical drive")
+
+        if len(self.disk_no) < int(self.disk_initial) or \
+           len(self.disk_no) < int(self.disk_migrated):
+            self.skip("Cannot Migrate, please check the prerequisite")
+
+    def test(self):
+        """
+        Main function
+        """
+        disk_val1 = ""
+        disk_val2 = ""
+        # create the inital disk required for raid creation
+        loop_count = 0
+        for disk in self.disk_no:
+            loop_count += 1
+            disk_val1 += " %s %s" % (self.channel_no, disk.strip())
+            if loop_count == 2 and self.initial_raid in ('0', '1'):
+                break
+            if loop_count == 3 and self.initial_raid == '5':
+                break
+            if loop_count == 4 and self.initial_raid in ('6', '10'):
+                break
+        # create the disk required for migration
+        loop_count = 0
+        for disk in self.disk_no:
+            loop_count += 1
+            disk_val2 += " %s %s" % (self.channel_no, disk.strip())
+            if loop_count == int(self.disk_migrated):
+                break
```

Comment:
@narasimhan-v  
with the for loops i set disk_val1 and disk_val2 which is the disk combination before and after the migration.  each of the variable needs to exit based on different test conditions.  hence wont be done.